src/Socket.js

```diff
@@ -54,7 +54,10 @@ export class Socket extends EventEmitter {
         this.connected = false
         this.emit('disconnected')
         if (this.opts.reconnect) {
-          this.reconnect()
+          this.reconnect().catch(() => {
+            let err = new Error(`Auto-reconnect failed after ${this.maxRetries} retries`)
```

Comment:
reconnect() already does throw new Error(`Too many connection failures ${this.maxRetries}`) and emits 'error', does this not just end up duplicating that?